lisp/ess-mode.el

```diff
@@ -1107,30 +1107,44 @@ Return the amount the indentation changed by."
           ;; 0))
           ((/= (char-after containing-sexp) ?{)
            ;; line is expression, not statement:
-           ;; indent to just after the surrounding open.
-           (goto-char containing-sexp)
-           (let ((bol (line-beginning-position)))
-
-             (cond ((and (numberp ess-expression-offset)
-                         (re-search-backward "[ \t]*expression[ \t]*(" bol t))
-                    ;; obj <- expression(...
-                    ;; modified by shiba (forward-sexp -1)
-
-                    (beginning-of-line)
-                    (skip-chars-forward " \t")
-                    ;; end{modified}
-                    (+ (current-column) ess-expression-offset))
-                   ((and (numberp ess-arg-function-offset)
-                         (re-search-backward
-                          "=[ \t]*\\s\"*\\(\\w\\|\\s_\\)+\\s\"*[ \t]*"
-                          bol t))
-                    (forward-sexp -1)
-                    (+ (current-column) ess-arg-function-offset))
-                   (t
+           ;; if we have a succession of blocks as arguments,
+           ;; don't indent with the surrounding open
+           (if (and (setq last-sexp (nth 2 state))
                     (or
-                     (ess-calculate-indent--after-last-open-paren)
-                     (progn (goto-char (1+ containing-sexp))
-                            (current-column)))))))
+                     (eq (char-after last-sexp) ?{)
+                     (and
+                      ;; in case we have additional non-block arguments
+                      (eq (char-before last-sexp) ?\C-j)
```

Comment:
I think it's a perfect name.

About the testing scheme, I'm trying to set up different R files containing the same testing set but with different configurations. To achieve this I tried to specify different file-local variables with the indentation settings. This doesn't work and indeed in the source code of `(ess-mode)` we have this `(kill-all-local-variables)` nuking all the variables I'm trying to set (IIUC).

I'm now trying to hack something by setting locally a modified `ess-local-customize-alist`, but is there a canonical way to do this?